src/fortran_routing/diffusive_v02frwk/diffusive_cnt/diffusive_cnt.f90

```diff
@@ -293,172 +296,143 @@ subroutine diffnw(dtini_g, t0_g, tfin_g, saveinterval_ev_g, dt_ql_g, dt_ub_g, dt
+					newQ(1,timestep,j) = 0.0
+					do k=1, frnw_g(j,3)
+						linknb= frnw_g(j,3+k)
+						nodenb= frnw_g(linknb,1)
+						! sum of junction inflows
+						newQ(1,timestep,j) = newQ(1,timestep,j) + newQ(nodenb,timestep,linknb)
+					end do
+				end if
+				! +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+				! populate lateral inflow array : lateralFlow in m2/sec
+				! +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+                    		do i=1,ncomp-1
 				    	!* 1. linear interpolation between qlat_g values
 !				    	do n=1,nts_ql_g
 !				    	    varr_ql(n)= qlat_g(n,i,j) !* qlat_g(n,i,j) in unit of m2/sec
 !				    	enddo
 !				    	tc_cnt= ini_time+dtini/60.0*real(timestep-1) !*[min]
-!                        lateralFlow(i, timestep, j)= intp_y(nts_ql_g, tarr_ql, varr_ql, tc_cnt)
-                        !* 2. stair case form of qlat_g (i.e., constant qlat over 60 min window)
-                        tc_cnt= ini_time+dtini/60.0*real(timestep-1) !*[min]
-                        idxql= locate(tarr_ql,tc_cnt)
-                        if (idxql.lt.1) idxql=1
-                        if (idxql.gt.nts_ql_g) idxql=nts_ql_g
-                        lateralFlow(i, timestep, j)= qlat_g(idxql,i,j)
-                    end do
-                    !* test start
-!                    tc_cnt= ini_time+real(timestep-1)*dtini/60.
-!                    write(10, *) tc_cnt, j, (lateralFlow(i,timestep,j), i=1, ncomp-1)
-                    !* test end
-
+!                        		lateralFlow(i, timestep, j)= intp_y(nts_ql_g, tarr_ql, varr_ql, tc_cnt)
+                        		!* 2. stair case form of qlat_g (i.e., constant qlat over 60 min window)
+                        		tc_cnt= ini_time+dtini/60.0*real(timestep-1) !*[min]
+                        		idxql= locate(tarr_ql,tc_cnt)
+                        		if (idxql.lt.1) idxql=1
+                        		if (idxql.gt.nts_ql_g) idxql=nts_ql_g
+                        		lateralFlow(i, timestep, j)= qlat_g(idxql,i,j)
+                    		end do
          			!* head water reach
-                    !if (frnw_g(j,3)==0) then
-                        newQ(1,timestep,j) = newQ(1,timestep,j)+lateralFlow(1,timestep,j)*dx(1,j)
-                        lateralFlow(1,timestep,j) = 0.0       !
-                    !endif
-!                    print*, "time", j, timestep, dtini, ini_time, tc_cnt
-				end do !* do timestep=1, ntim
-				!lateralFlow(1,:,j) = 0.       ! lateral flow at i = 1 is already added
-
-				! +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-				! Diffusive wave forward sweep to calculate flow at t+1
-				! +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-				qp(:,:,j) = newQ(:,:,j)
-
-				call diffusive_CNT(j,ntim,repeatInterval)
-	
-				newQ(:,:,j) = qp(:,:,j)
-				
-			end do  !* j = 1, nlinks
+                           	newQ(1,timestep,j) = newQ(1,timestep,j)+lateralFlow(1,timestep,j)*dx(1,j)
```

Comment:
@kumdonoaa should we change this comment? This is not a headwater reach, rather it is the upstream most node of reach j. Here, we are adding the lateral inflows into newQ at that node.